opm/grid/LookUpData.hh

```diff
@@ -377,6 +444,41 @@ Opm::LookUpCartesianData<Grid,GridView>::operator()(const EntityType& elem,
     return fieldProp[fieldPropCartIdx];
 }
 
+template<typename Grid, typename GridView>
+std::vector<double> Opm::LookUpCartesianData<Grid,GridView>::assignFieldPropsDoubleOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                                          const std::string& propString,
+                                                                                          const unsigned int& numElements,
+                                                                                          const double& defaultValue) const
+{
+    std::vector<double> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
+    const auto& fieldProp = fieldPropsManager.get_double(propString);
+    for (unsigned int elemIdx = 0; elemIdx < numElements; ++elemIdx) {
+        const auto fieldPropCartIdx = this->getFieldPropCartesianIdx<Grid>(elemIdx);
+        fieldPropOnLeaf[elemIdx] = fieldProp[fieldPropCartIdx];
+    }
+    return fieldPropOnLeaf;
+}
+
+template<typename Grid, typename GridView>
+template<typename IntType>
+std::vector<IntType> Opm::LookUpCartesianData<Grid,GridView>::assignFieldPropsIntOnLeaf(const FieldPropsManager& fieldPropsManager,
+                                                                                        const std::string& propString,
+                                                                                        const unsigned int& numElements,
+                                                                                        const int& defaultValue,
+                                                                                        const bool& needsTranslation,
+                                                                                        std::function<void(IntType)> valueCheck) const
+{
+    std::vector<IntType> fieldPropOnLeaf;
+    fieldPropOnLeaf.resize(numElements, defaultValue);
+    const auto& fieldProp = fieldPropsManager.get_int(propString);
+    for (unsigned elemIdx = 0; elemIdx < numElements; ++elemIdx) {
+        const auto fieldPropCartIdx = this->getFieldPropCartesianIdx<Grid>(elemIdx);
+        fieldPropOnLeaf[elemIdx] = fieldProp[fieldPropCartIdx] - needsTranslation;
+        valueCheck(fieldPropOnLeaf[elemIdx]);
+    }
+    return fieldPropOnLeaf;
```

Comment:
Same comments here:
Should we translate the default value?
Should we check with the original value?

Reply:
I didn't get the translated defualt value, but it seems I could get the default value with the property name, via an unordered map. See my comment on OPM/opm-simulators#4951